### PR TITLE
Improve design of the rich text editor

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

What it looks like:

https://user-images.githubusercontent.com/2621378/199945374-f0495971-5aee-4d5b-957a-9effea6a08aa.mov

Few things here:
- To avoid custom margin under and between each button, I wrapped and centered them in a container
     - I used `gap` for the new and current composer for the space between the buttons
     - For the new composer, the container is placed in order to have always the same position sticked to the bottom even when the the composer is growing
- The emoji button is inside the composer in the new composer, I extracted it from `MessageComposerButtons` to use it in both cases. I created a CSS mixin to share the style between the two use cases. 
- I had to reduce the button margin in the `VoiceRecordComposer` component because there is already the `gap` between them
- The `VoiceRecordComposer` padding is reduced in the new composer to have the same height than the composer (ie: `40px`)
- The plain / rich text icons were inverted
- The plain text icon had some issues when highlighted, the icon is reworked
- The border radius is different when the composer has 1 line or multiples lines of content (hook `useIsExpanded` using a ResizeObserver)
- The border colour changes when the editor has the focus. I had to check manually the focus (not using the CSS pseudo class `focused`) because the border is not on the focused element but in one of the parents
- Fix TS strict error as well in the modified files

---
Here's what your changelog entry will look like:

## ✨ Features
 * Improve design of the rich text editor ([\#9533](https://github.com/matrix-org/matrix-react-sdk/pull/9533)). Contributed by @florianduros.

### CODE REVIEW

File: src/components/views/rooms/wysiwyg_composer/hooks/useIsExpended.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MutableRefObject, useEffect, useState } from "react";
+
+export function useIsExpanded(ref: MutableRefObject<HTMLElement>, breakingPoint: number) {
```
Comment (t3chguy) [inline src/components/views/rooms/wysiwyg_composer/hooks/useIsExpended.ts:19]: file name doesn't match export name

File: res/css/views/rooms/_EmojiButton.pcss [hunk 2]
```diff
@@ -0,0 +1,13 @@
+@import "./_MessageComposerButton.pcss";
```
Comment (t3chguy) [inline res/css/views/rooms/_EmojiButton.pcss:1]: missing copyright header

File: res/css/views/rooms/_MessageComposerButton.pcss [hunk 3]
```diff
@@ -0,0 +1,42 @@
+@define-mixin composerButtonHighLight {
```
Comment (t3chguy) [inline res/css/views/rooms/_MessageComposerButton.pcss:1]: missing copyright header

File: test/components/views/rooms/wysiwyg_composer/components/PlainTextComposer-test.tsx [hunk 4]
```diff
@@ -91,4 +87,40 @@ describe('PlainTextComposer', () => {
...
+            };
+        },
+        );
+
+        //When
+        render(
+            <PlainTextComposer onChange={jest.fn()} onSend={jest.fn()} />,
+        );
+
+        console.log(screen.getByTestId('WysiwygComposerEditor'), screen.getByTestId('WysiwygComposerEditor').dataset);
```
Comment (robintown) [inline test/components/views/rooms/wysiwyg_composer/components/PlainTextComposer-test.tsx:112]: Is this left over from debugging? Logs tend to slow down tests

File: src/components/views/rooms/MessageComposerButtons.tsx [hunk 5]
```diff
@@ -75,7 +77,9 @@ const MessageComposerButtons: React.FC<IProps> = (props: IProps) => {
     let moreButtons: ReactElement[];
     if (narrow) {
         mainButtons = [
-            emojiButton(props),
+            !isWysiwygLabEnabled && emojiButton(props),
+            isWysiwygLabEnabled &&
+                <ComposerModeButton key="composerModeButton" isRichTextEnabled={props.isRichTextEnabled} onClick={props.onComposerModeClick} />,
```
Comment (robintown) [inline src/components/views/rooms/MessageComposerButtons.tsx:82]:
```suggestion
            isWysiwygLabEnabled
                ? <ComposerModeButton key="composerModeButton" isRichTextEnabled={props.isRichTextEnabled} onClick={props.onComposerModeClick} />
                : emojiButton(props),
```

File: src/components/views/rooms/MessageComposerButtons.tsx [hunk 6]
```diff
@@ -87,8 +91,8 @@ const MessageComposerButtons: React.FC<IProps> = (props: IProps) => {
         ];
     } else {
         mainButtons = [
-            emojiButton(props),
-            props.showComposerModeButton &&
+            !isWysiwygLabEnabled && emojiButton(props),
+            isWysiwygLabEnabled &&
                 <ComposerModeButton key="composerModeButton" isRichTextEnabled={props.isRichTextEnabled} onClick={props.onComposerModeClick} />,
```
Comment (robintown) [inline src/components/views/rooms/MessageComposerButtons.tsx:96]: This would also be more readable as a ternary

**Review by robintown — COMMENTED** [review]
It'd be nice to see a screenshot!

**Review by robintown — APPROVED** [review]
Looks fantastic

File: cypress/support/composer.ts [hunk 7]
```diff
@@ -32,6 +32,14 @@ declare global {
     }
 }
 
+const resizeObserverLoopErrRe = /^[^(ResizeObserver loop limit exceeded)]/;
+Cypress.on('uncaught:exception', (err) => {
+    /* returning false here prevents Cypress from failing the test */
+    if (resizeObserverLoopErrRe.test(err.message)) {
+        return false;
+    }
+});
```
[the remark this answers is not in the document]
Comment (florianduros) [reply cypress/support/composer.ts]: I removed it and added a requestAnimationFrame to avoid the error :)

**Pull request merged**